### PR TITLE
Fix pre commit issues

### PR DESCRIPTION
**Pull request by ty338jy**
@yuanjingshi @ps2016-mt 

Hello guys, please review the fix for pre-commit run, fixes including:
- line of codes should have characters less than 78
- functions should be 2 blank lines apart from each other
- mypy is running issue with request (missing library stubs)
1. mypy.ini was created to set ignore_missing_imports =True
2. pre-commit-config.yaml was updated

### CODE REVIEW

File: mypy.ini
Comment (yuanjingshi) [inline mypy.ini:1]: Maybe to add a comment of why ignore imports missing.

**Review by yuanjingshi — CHANGES_REQUESTED** [review]

File: mypy.ini
Comment (ty338jy) [reply mypy.ini:1]: Hey Yuanjing,
Comments added as requested!

**Review by yuanjingshi — DISMISSED** [review]
Looks very good. I like your clear description in the PR. I shall learn from you on that.

**Comment by ps2016-mt** [comment]
agreed, looks good!

**Review by ps2016-mt — DISMISSED** [review]
agreed, looks good!

**Review by ps2016-mt — DISMISSED** [review]
merge base seems fine with me

**Review by cameds58 — APPROVED** [review]

**Review by cameds58 — DISMISSED** [review]

**Review by yuanjingshi — DISMISSED** [review]

**Review by jps215 — APPROVED** [review]
Looks good

**Review by yuanjingshi — DISMISSED** [review]

**Review by cameds58 — DISMISSED** [review]

**Pull request merged**